scripts/push_l10n_extraction.sh

```diff
@@ -9,38 +9,50 @@ set -o pipefail
 # Treat unset variables as an error an exit immediately.
 set -u
 
+set -x
+
+info() {
+  local message="$1"
+
+  echo ""
+  echo "INFO: $message"
+  echo ""
+}
+
 ROBOT_EMAIL="addons-dev-automation+github@mozilla.com"
 ROBOT_NAME="Mozilla Add-ons Robot"
 
+# Set git committer/author to the robot.
+export GIT_AUTHOR_NAME="$ROBOT_NAME"
+export GIT_AUTHOR_EMAIL="$ROBOT_EMAIL"
+export GIT_COMMITTER_NAME="$ROBOT_NAME"
+export GIT_COMMITTER_EMAIL="$ROBOT_EMAIL"
+
 DATE=$(date -u +%Y-%m-%d)
 REV=$(git rev-parse --short HEAD)
 MESSAGE="Extracted l10n messages from $DATE at $REV"
 DIFF_WITH_ONE_LINE_CHANGE="2 files changed, 2 insertions(+), 2 deletions(-)"
 
-DRY_RUN=true
-
-if [[ "${1:-}" != "--dry-run" ]]; then
-  DRY_RUN=false
-fi
-
 git_diff_stat=$(git diff --shortstat locale/templates/LC_MESSAGES)
 
-echo "git_diff_stat: $git_diff_stat"
+info "git_diff_stat: $git_diff_stat"
 
 # IF there are no uncommitted local changes, exit early.
 if [[ -z "$git_diff_stat" ]] || [[ "$git_diff_stat" == *"$DIFF_WITH_ONE_LINE_CHANGE"* ]]; then
-  echo """
+  info """
     No substantial changes to l10n strings found. Exiting the process.
   """
   exit 0
 fi
 
-if [[ $DRY_RUN == true ]]; then
-  echo "Dry running..."
-  echo "git commit --author=\"$ROBOT_NAME <$ROBOT_EMAIL>\" -a -m \"$MESSAGE\""
-  echo "git push"
-else
-  # Commit the changes and push them to the repository.
-  git commit --author="$ROBOT_NAME <$ROBOT_EMAIL>" -a -m "$MESSAGE"
-  git push
-fi
+info """
+GIT_AUTHOR_NAME: $GIT_AUTHOR_NAME
+GIT_AUTHOR_EMAIL: $GIT_AUTHOR_EMAIL
+GIT_COMMITTER_NAME: $GIT_COMMITTER_NAME
+GIT_COMMITTER_EMAIL: $GIT_COMMITTER_EMAIL
+
+ARGS: $@
+"""
+
+git commit -am "$MESSAGE" "$@"
+git push "$@"
```

Comment:
This is going to avoid committing because of the `dry-run`, so even without the `dry-run` the push wouldn't work, as there would be nothing to push (the commit never happened for real). The end result is the same I guess though...

Reply:
I think it's fine to keep it consistent. The one downside with this is the acceptable flags for commit and push overlap but not entirely so you can get some weird end states.

I don't think we will play around with this much it's mostly just the dry-mode we want, it's just easier to pass directly because it is literally the same flag and already does what we want even better than before where we just logged. dry mode on the command actually verifies it "would have" worked.